spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -0,0 +1,490 @@
+#!/usr/bin/env python
+#########################################################################################
+#
+# Compute maximum spinal cord compression using AP diameter or other morphometrics.
+#
+# ---------------------------------------------------------------------------------------
+# Copyright (c) 2015 Polytechnique Montreal <www.neuro.polymtl.ca>
+# Author: Sandrine Bédard, Jan Valosek, Julien Cohen-Adad
+#
+# About the license: see the file LICENSE.TXT
+#########################################################################################
+# TODO: add option to normalize or not
+# TODO: maybe create an API or move some functions
+import sys
+import os
+import numpy as np
+import csv
+import logging
+from typing import Sequence
+import pandas as pd
+from spinalcordtoolbox.utils import SCTArgumentParser, Metavar, init_sct, set_loglevel
+from spinalcordtoolbox.utils.fs import get_absolute_path, check_file_exist, extract_fname, printv
+from spinalcordtoolbox.image import Image
+from spinalcordtoolbox import __data_dir__
+
+
+logger = logging.getLogger(__name__)
+
+
+# PARSER
+# ==========================================================================================
+def get_parser():
+    parser = SCTArgumentParser(
+        description='Compute spinal cord compression morpometrics such as Maximum Spinal Cord Compression as in'
+                    'Miyanji F, Furlan JC, Aarabi B, Arnold PM, Fehlings MG. Acute cervical traumatic spinal cord injury:'
+                    ' MR imaging findings correlated with neurologic outcome--prospective study with 100 consecutive patients.'
+                    ' Radiology 2007;243(3):820-827.'
+                    'Can be computed using AP diameter or other morphometrics (CSA, RL diameter, eccentricity and solidity).'
+                    'Morphometrics are normalized with a reference database where morphometrics from sct_process_segmentation '
+                    'are brought to PAM50 anatomical space using -normalize-PAM50.'
+    )
+
+    mandatoryArguments = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatoryArguments.add_argument(
+        '-i',
+        metavar=Metavar.file,
+        required=True,
+        help='Input csv file name in native space. Example: csa.csv.'
+    )
+    mandatoryArguments.add_argument(
+        '-l',
+        metavar=Metavar.file,
+        required=True,
+        help='.nii file with compression labels. Each compression is denoted by a single voxel of value `1`. '
+             'Example: sub-001_T2w_compression_labels.nii.gz'
+    )
+    mandatoryArguments.add_argument(
+        '-i-PAM50',
+        metavar=Metavar.file,
+        required=True,
+        help='Input csv file name in PAM50 space. Example: csa_PAM50.csv.'
+        'Use sct_process_segmentation with -normalize-PAM50 to generate the file.'
+    )
```

Comment:
Do we want to have it mandatory? What if users want to compute MSCC without the normalization (ie: the traditional way)?